app/src/main/java/com/codingblocks/cbonlineapp/viewmodels/VideoPlayerViewModel.kt

```diff
@@ -0,0 +1,77 @@
+package com.codingblocks.cbonlineapp.viewmodels
+
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.codingblocks.cbonlineapp.database.CourseDao
+import com.codingblocks.cbonlineapp.database.CourseRunDao
+import com.codingblocks.cbonlineapp.database.DoubtsDao
+import com.codingblocks.cbonlineapp.database.NotesDao
+import com.codingblocks.cbonlineapp.database.models.DoubtsModel
+import com.codingblocks.cbonlineapp.database.models.NotesModel
+import com.codingblocks.cbonlineapp.extensions.retrofitCallback
+import com.codingblocks.onlineapi.Clients
+import com.codingblocks.onlineapi.models.DoubtsJsonApi
+import com.codingblocks.onlineapi.models.Notes
+import java.lang.Exception
+
+class VideoPlayerViewModel(
+    var doubtsDao: DoubtsDao,
+    var notesDao: NotesDao,
+    var courseDao: CourseDao,
+    var runDao: CourseRunDao
+) : ViewModel() {
+    var playWhenReady = false
+    var currentOrientation: Int = 0
+    var mOtp: String? = null
+    var mPlaybackInfo: String? = null
+
+    var getOtpProgress: MutableLiveData<Boolean> = MutableLiveData()
+    var createDoubtProgress: MutableLiveData<Boolean> = MutableLiveData()
+    var createNoteProgress: MutableLiveData<Boolean> = MutableLiveData()
+
+    fun getOtp(videoId: String, sectionId: String, attemptId: String) {
+        Clients.api.getOtp(videoId, sectionId, attemptId)
+            .enqueue(retrofitCallback { _, response ->
+                response?.let {
+                    mOtp = response.body()?.get("otp")?.asString
+                    mPlaybackInfo = response.body()?.get("playbackInfo")?.asString
+                    getOtpProgress.value = (it.isSuccessful)
+                }
+            })
+    }
+
+    fun createDoubt(doubt: DoubtsJsonApi) {
+        Clients.onlineV2JsonApi.createDoubt(doubt)
+            .enqueue(retrofitCallback { _, response ->
+                doubtsDao.insert(
+                    DoubtsModel(
+                        doubt.id, doubt.title, doubt.body, doubt.contents?.id
+                        ?: "", doubt.status, doubt.runAttempt?.id ?: ""
+                    )
+                )
+                createDoubtProgress.value = (response?.isSuccessful == true)
+            })
+    }
+
+    fun createNote(note: Notes) {
+        Clients.onlineV2JsonApi.createNote(note)
+            .enqueue(retrofitCallback { _, response ->
+                try {
```

Comment:
why only notes are in try catch block and not others ?

Reply:
It is because the original file had a try block in notes, and others didn't

Reply:
I've updated the rest dao operations as well :)

Reply:
i made some changes in this pr why had you removed them?

Reply:
@aggarwalpulkit596 Must have occured because I force pushed the branch, can you suggest the changes again?

Reply:
you're inserting notes and doubt irrespective of the response is successful or not it should be something like `if(response.isSuccesful)`